src/apps/tools/private.clj

```diff
@@ -9,6 +10,14 @@
             [apps.util.config :as cfg]
             [apps.validation :as validation]))
 
+(defn- validate-image-not-deprecated
+  [image-info]
+  (let [image (containers/find-image-by-name-and-tag image-info)]
+    (when (:deprecated image)
+    (throw+ {:type  :clojure-commons.exception/bad-request-field
+             :error "Image is deprecated and should not be used in new tools."
+             :image image}))))
```

Comment:
This `throw+` call might be another good candidate for `clojure-commons.exception-util`. It doesn't save that much space, but I like to use those functions for convenience just because I'm lazy and it saves me typing. 😆